src/target/linux.rs

```diff
@@ -268,4 +316,60 @@ mod tests {
             dealloc(ptr, layout);
         }
     }
+
+    #[test]
+    fn reads_threads() -> Result<(), Box<dyn std::error::Error>> {
+        let start_barrier = Arc::new(Barrier::new(2));
+        let end_barrier = Arc::new(Barrier::new(2));
+
+        let t1_start = start_barrier.clone();
+        let t1_end = end_barrier.clone();
+
+        let thread_name = "thread_name";
+        let t1_handle = thread::Builder::new()
+            .name(thread_name.to_string())
+            .spawn(move || {
+                t1_start.wait();
+                t1_end.wait();
+            })
+            .unwrap();
+
+        start_barrier.wait();
+
+        let proc = LinuxTarget::me();
+        let threads = proc.threads()?;
+
+        let threads: Vec<_> = threads
+            .iter()
+            .map(|t| (t.name().unwrap().clone(), t.thread_id()))
+            .collect();
+
+        // Find at least 3 threads (cargo runs a minimum of 2)
```

Comment:
> cargo runs a minimum of 2

This is not always the case and it's configurable using the `test-threads` parameter (e.g.: `cargo test -- --test-threads=1`) or the environment variable `RUST_TEST_THREADS`.

As it seems, currently there's no way to get a number of test threads from _inside_ the tests (reading the env var is inconsistent), but I guess for the moment we can do something like

```suggestion
        // Find at least 3 threads (cargo runs a minimum of 2)
        let cargo_threads = std::env::var("RUST_TEST_THREADS")
            .map(|s| s.parse::<usize>())
            .unwrap_or(2);
```

Reply:
When I considered cargo 2 threads it was the main thread + the test thread. 
Still your comment applies, I will do that +1 :)

Reply:
[Reference](https://stackoverflow.com/questions/43458194/is-there-any-way-to-tell-cargo-to-run-its-tests-on-the-main-thread)

Reply:
should be fixed now